lms/djangoapps/courseware/views/index.py

```diff
@@ -415,6 +421,29 @@ def _create_courseware_context(self, request):
             (settings.FEATURES.get('ENABLE_COURSEWARE_SEARCH_FOR_COURSE_STAFF') and self.is_staff)
         )
 
+        reset_deadlines_url = reverse(
+            'openedx.course_experience.reset_course_deadlines', kwargs={'course_id': six.text_type(self.course.id)}
+        )
+
+        display_reset_dates_banner = False
+
+        course_overview = CourseOverview.objects.get(id=str(self.course_key))
+        if timezone.now() < course_overview.end_date:
+            course_enrollment = CourseEnrollment.objects.filter(course=course_overview, user=request.user).first()
+            if course_enrollment and course_enrollment.mode == CourseMode.VERIFIED:
+                course_block_tree = get_course_outline_block_tree(
+                    request, str(self.course_key), request.user
+                )
+                course_sections = course_block_tree.get('children')
+                for section in course_sections:
+                    if display_reset_dates_banner:
+                        break
+                    for subsection in section.get('children', []):
+                        if not subsection.get('complete', True)\
+                                and subsection.get('due', timezone.now() + timedelta(1)) < timezone.now():
+                            display_reset_dates_banner = True
+                            break
```

Comment:
Should this whole bit be broken off into some utility function if we think we might need to be repeating this logic on the course home and in the courseware?